app/assets/javascripts/newflow/index.js

```diff
@@ -36,6 +36,26 @@ $(document).ready(function(){
     }
   });
 
+  // Validate name for unallowed chars
+  var errorDiv = '<div class="errors invalid-message">Name cannot contain special characters</div>';
+  function check_for_specialness(obj) {
+      if(/^[a-zA-Z0-9- ]*$/.test(obj.value) == false) {
+          $(obj).addClass('has-error');
+          $(obj).closest('.control-group').append(errorDiv);
+      }else{
+          $(obj).removeClass('has-error');
+          $(obj).closest('.control-group').remove(errorDiv);
+      }
+  }
+  $("#signup_first_name").blur(function (e) {
+      check_for_specialness(this);
+  });
+
+    $("#signup_last_name").blur(function (e) {
+        check_for_specialness(this);
+    });
```

Comment:
Honestly, I wasn't even going to add an error message to the UI, just do the removal (because people should know better!). If I get time to look at this, I can attempt.

Reply:
Ok, that's alright. But I think I would actually vote for removing the frontend validation if it isn't going to be feature tested. Fyi.